test/Remapping/distributed_remapping.jl

```diff
@@ -0,0 +1,308 @@
+using Logging
+using Test
+using IntervalSets
+
+import ClimaCore:
+    Domains, Fields, Geometry, Meshes, Operators, Spaces, Topologies, Remapping
+using ClimaComms
+const context = ClimaComms.MPICommsContext()
+const pid, nprocs = ClimaComms.init(context)
+const device = ClimaComms.CPUSingleThreaded()
+
+# log output only from root process
+logger_stream = ClimaComms.iamroot(context) ? stderr : devnull
+prev_logger = global_logger(ConsoleLogger(logger_stream, Logging.Info))
+atexit() do
+    global_logger(prev_logger)
+end
+
+@testset "2D extruded" begin
+    vertdomain = Domains.IntervalDomain(
+        Geometry.ZPoint(0.0),
+        Geometry.ZPoint(1000.0);
+        boundary_tags = (:bottom, :top),
+    )
+
+    vertmesh = Meshes.IntervalMesh(vertdomain, nelems = 30)
+    verttopo = Topologies.IntervalTopology(
+        ClimaComms.SingletonCommsContext(device),
+        vertmesh,
+    )
+    vert_center_space = Spaces.CenterFiniteDifferenceSpace(verttopo)
+
+    horzdomain = Domains.IntervalDomain(
+        Geometry.XPoint(-500.0) .. Geometry.XPoint(500.0),
+        periodic = true,
+    )
+
+    quad = Spaces.Quadratures.GLL{4}()
+    horzmesh = Meshes.IntervalMesh(horzdomain, nelems = 10)
+    horztopology = Topologies.IntervalTopology(
+        ClimaComms.SingletonCommsContext(device),
+        horzmesh,
+    )
+    horzspace = Spaces.SpectralElementSpace1D(horztopology, quad)
+
+    hv_center_space =
+        Spaces.ExtrudedFiniteDifferenceSpace(horzspace, vert_center_space)
+
+    coords = Fields.coordinate_field(hv_center_space)
+
+    xpts = range(-500.0, 500.0, length = 21)
```

Comment:
I think you should be able to just do
```suggestion
    xpts = range(-Geometry.XPoint(500.0), Geometry.XPoint(500.0), length = 21)
```

Reply:
I can do this only in the case of a plane. 

I need to build `hcoords` with the correct type, and I cannot easily assemble, e.g.,  `XYPoint`s from `XPoint`s and `YPoint`s (see https://github.com/CliMA/ClimaCore.jl/issues/1476).